src/rust/engine/src/nodes/scandir.rs

```diff
@@ -20,17 +21,40 @@ pub struct Scandir {
     pub(super) dir: Dir,
     pub(super) subject_path: SubjectPath,
 }
-
 impl Scandir {
     pub(super) async fn run_node(self, context: Context) -> NodeResult<Arc<DirectoryListing>> {
+        let gitignore_stack = self.parent_gitignore_stack(&context).await?;
         let directory_listing = context
             .core
             .vfs
-            .scandir(self.dir)
+            .scandir(self.dir, gitignore_stack)
             .await
             .map_err(|e| throw(format!("{e}")))?;
         Ok(Arc::new(directory_listing))
     }
+
+    async fn parent_gitignore_stack(&self, context: &Context) -> NodeResult<GitignoreStack> {
+        let root_ignore = context.core.vfs.root_ignore();
+        if !context.core.vfs.read_gitignore_files() {
```

Comment:
I'm confused: As far as I can tell, this will be true iff `use_gitignore` is true (in `Core::new()`). But if it's false shouldn't we be using no gitignores at all, not even the root one?

Reply:
Ah, that is non-obvious. So how are `pants_ignore` and root `gitignore` combined?

Reply:
"pants_ignore", including allowlisting, is always applied first. Then git-style stack traversal follows. This is equivalent to the previous behaviour, we just previously had a stack of depth=1.

Reply:
Yeah, that would make it clearer.